### ISSUE TITLE
Problems initializing the Jep object and importing libraries after upgrading to Jep 3.9.0

### ISSUE THREAD

**Issue by Melnorme1984** [ninia/jep#208]
My application which previously worked with Jep 3.8.2 has broken after upgrading to 3.9.0.

I'm actually trying to run it in two different environments and getting two different errors. Both using Java 8 and Python 2.7.x. 

This is how I initialize the Jep object:

```
    private static final ExecutorService service = Executors.newSingleThreadExecutor();
    private static Interpreter jep;

    public static void init() {
        service.submit(() -> {
            try {
                jep = new SubInterpreter();
                jep.runScript(ModulomicsConfiguration.getModulomicsDir() + FileSystems.getDefault().getSeparator() + ModulomicsConfiguration.getPlotterFile());
                jep.invoke("init", ModulomicsConfiguration.getDataDir());
            } catch (JepException ex) {
                logger.log(Level.SEVERE, null, ex);
            }
        });
    }
```

The runScript line initializes a script that starts like this:

```
import collections
import matplotlib
matplotlib.use("Agg")
import matplotlib.gridspec as gridspec
import matplotlib.pyplot as pyplot
import mpld3
import os
import pandas
import seaborn
```

On Linux, which is my primary environment, it never even gets past the initialization of the object. This is the error I'm getting:

```
	jep.JepException: <type 'exceptions.RuntimeError'>: getPyObject
		at jep.Jep.set(Native Method)
		at jep.Jep.set(Jep.java:762)
		at jep.Jep.setupJavaImportHook(Jep.java:297)
		at jep.Jep.configureInterpreter(Jep.java:285)
		at jep.Jep.<init>(Jep.java:256)
		at jep.Jep.<init>(Jep.java:228)
		at jep.Jep.<init>(Jep.java:140)
		at jep.SubInterpreter.<init>(SubInterpreter.java:58)
		at modulomics.server.PythonInterface.lambda$init$0(PythonInterface.java:34)
		at modulomics.server.PythonInterface$$Lambda$24/1906582038.run(Unknown Source)
		at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
		at java.util.concurrent.FutureTask.run(FutureTask.java:266)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
		at java.lang.Thread.run(Thread.java:745)
	Caused by: java.lang.NoSuchMethodError: getPyObject
		... 15 more
```

On my personal Windows machine, it gets a bit further but for some reason fails to import the various Python libraries from the script, with errors such as this:

```
jep.JepException: <type 'exceptions.ImportError'>: cannot import name _remove_dead_weakref
	at C:\Users\yomtova\AppData\Local\Continuum\anaconda2\lib\weakref.<module>(weakref.py:14)
	at C:\Users\yomtova\AppData\Local\Continuum\anaconda2\lib\logging\__init__.<module>(__init__.py:26)
	at C:\Users\yomtova\AppData\Local\Continuum\anaconda2\lib\site-packages\matplotlib\__init__.<module>(__init__.py:115)
	at D:\Tovi\ModulOmics\ModulOmics_webserver.<module>(ModulOmics_webserver.py:2)
	at jep.Jep.run(Native Method)
	at jep.Jep.runScript(Jep.java:384)
	at jep.Jep.runScript(Jep.java:355)
	at modulomics.server.PythonInterface.lambda$init$0(PythonInterface.java:35)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

These are the relevant environment variables on Linux:

```
JAVA_HOME=/usr/local/lib/jdk-8u25/
LD_LIBRARY_PATH=/usr/local/stow/jdk-8u25/lib/jdk-8u25/jre/lib/amd64/server:/usr/local/lib/python2.7/dist-packages/jep
LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libpython2.7.so
```

And on Windows:

```
JAVA_HOME=C:\Program Files\Java\jdk1.8.0_231\
JRE_HOME=C:\Program Files\Java\jre1.8.0_231\
PATH=C:\Users\yomtova\AppData\Local\Continuum\anaconda2\Lib\site-packages\jep
PYTHONHOME=C:\Users\yomtova\AppData\Local\Continuum\anaconda2
```

Any idea? Like I said, it worked before. I tried using a SharedInterpreter instead of a SubInterpreter on Windows but it didn't change anything.

**Comment by bsteffensmeier** [comment]
The getPyObject method was added in Jep 3.9.0. Since it is not found then I suspect you are using a jep 3.9 native library with a jep 3.8 java jar. Interestingly you are using the SubInterpreter class, which did not exist in 3.8, so you must have the 3.9 jar but the line numbers on for the Jep.java in your stack trace line up with 3.8, so the 3.8 jar is probably also on your class path and taking precedence.

The windows error seems to be related to python issue 28247, and may be occuring if your environment has more then one python installed and is mixing libraries from the two, I would recommend checking your environment carefully to make sure the correct python is being used.

**Comment by Melnorme1984** [comment]
Ooh yeah, you're right about the JAR. I should have caught that myself, it's happened to me before. Chalk it up to getting distracted by the other error.

I'm pretty sure I don't have more than one version of Python on my Windows machine, but I guess I'll take a look next week and see if there's anything weird going on in there.

**Comment by chenbekor** [comment]
@bsteffensmeier can you please further elaborate on how to resolve this issue?
I'm getting a similar exception:

`Caused by: jep.JepException: <class 'RuntimeError'>: getPyObject
	at jep.Jep.set(Native Method)
	at jep.Jep.set(Jep.java:762)
	at jep.Jep.setupJavaImportHook(Jep.java:297)
	at jep.Jep.configureInterpreter(Jep.java:285)
	at jep.Jep.<init>(Jep.java:256)
	at jep.Jep.<init>(Jep.java:228)
	at me.shadaj.scalapy.py.JepInterpreter.<init>(JepInterpreter.scala:9)
	at me.shadaj.scalapy.py.Platform$.newInterpreter(Platform.scala:5)
	at me.shadaj.scalapy.py.package$.interpreter(package.scala:11)
	at me.shadaj.scalapy.py.Writer$$anon$11.write(Writer.scala:75)
Caused by: java.lang.NoSuchMethodError: getPyObject
	at jep.Jep.set(Native Method)
	at jep.Jep.set(Jep.java:762)
	at jep.Jep.setupJavaImportHook(Jep.java:297)
	at jep.Jep.configureInterpreter(Jep.java:285)
	at jep.Jep.<init>(Jep.java:256)
	at jep.Jep.<init>(Jep.java:228)
	at me.shadaj.scalapy.py.JepInterpreter.<init>(JepInterpreter.scala:9)
	at me.shadaj.scalapy.py.Platform$.newInterpreter(Platform.scala:5)
	at me.shadaj.scalapy.py.package$.interpreter(package.scala:11)
	at me.shadaj.scalapy.py.Writer$$anon$11.write(Writer.scala:75)`

**Comment by bsteffensmeier** [comment]
This error occurs when you have the native library for jep 3.9 and the jar for an older version of jep. You are not using a the jar from jep 3.9 or perhaps you have both a jep 3.9 and an earlier jep jar. The solution is to remove any jep jar older than 3.9 and change any dependency management that may be downloading an older jar jar and instead only use the jep 3.9 jar. Since you are using scalapy, you may be interested in the[ scalapy issue that occurred after the jep 3.9 upgrade](https://github.com/shadaj/scalapy/issues/33). I don't think any version of scalapy was ever released that is compatible with jep 3.9
